upstream_delineator/py/util.py

```diff
@@ -510,18 +512,19 @@ def save_network(G: networkx.Graph, prefix: str, file_ext: str):
 
     filename = f"{OUTPUT_DIR}/{prefix}_graph.{file_ext}"
 
-    match file_ext:
-        case 'pkl':
-            # (1) Python pickle file
-            pickle.dump(G, open(filename, "wb"))
-        case 'json':
-            # (2) JSON file
-            data = networkx.node_link_data(G)
-            with open(filename, "w") as f:
-                json.dump(data, f)
-        case 'gml':
-            # (3) GML (Graph Modeling Language), a common graph file format.
-            networkx.write_gml(G, filename)
-        case 'xml':
-            # (4) GraphML is an XML-based file format for graphs.
-            networkx.write_graphml(G, filename)
+    if file_ext == 'pkl':
+        # (1) Python pickle file
+        pickle.dump(G, open(filename, "wb"))
+    elif file_ext == 'json':
+        # (2) JSON file
+        data = networkx.node_link_data(G)
+        with open(filename, "w") as f:
+            json.dump(data, f)
+    elif file_ext == 'gml':
+        # (3) GML (Graph Modeling Language), a common graph file format.
+        networkx.write_gml(G, filename)
+    elif file_ext == 'xml':
+        # (4) GraphML is an XML-based file format for graphs.
+        networkx.write_graphml(G, filename)
+    else:
+        raise ValueError(f'Unhandled file extension {file_ext}')
```

Comment:
for a follow up - we'll never hit this bc of this bit of code above: 
```
    allowed_formats = ['pkl', 'gml', 'xml', 'json']
    if file_ext not in allowed_formats:
        print(f"River network graph not saved. Please choose one of the following formats: "
              f"{', '.join(allowed_formats)}")
        raise Warning("Did not save graph data.")
```

Reply:
I like having this here to handle the case where we add a new format to the `allowed_formats` but then forget to handle the new format in the `if elif else` conditional

Reply:
agreed! i was thinking nixing the allowed formats stuff above and centralizing it down here in the else block